catalog/ftp/src/main/java/ddf/catalog/ftp/FtpServerManager.java

```diff
@@ -236,24 +236,31 @@ private void waitForConnections() {
         ((DefaultFtpServer) server).getServerContext().getFtpStatistics();
 
     int totalWait = 0;
+    boolean interrupt = false;
 
-    while (serverStatistics.getCurrentConnectionNumber() > 0) {
-      LOGGER.debug(
-          "Waiting for {} connections to close before updating configuration",
-          serverStatistics.getCurrentConnectionNumber());
-      try {
-        if (totalWait <= maxSleepTimeMillis) {
-          totalWait += resetWaitTimeMillis;
-          Thread.sleep(resetWaitTimeMillis);
-        } else {
-          LOGGER.debug(
-              "Waited {} seconds for connections to close, updating FTP configuration",
-              TimeUnit.MILLISECONDS.toSeconds(totalWait));
-          break;
+    try {
+      while (serverStatistics.getCurrentConnectionNumber() > 0) {
+        LOGGER.debug(
+            "Waiting for {} connections to close before updating configuration",
+            serverStatistics.getCurrentConnectionNumber());
+        try {
+          if (totalWait <= maxSleepTimeMillis) {
```

Comment:
❓  The potential problem I see here is that if we are interrupted, we loop back and sleep again for the same amount of time. The amount of time to sleep should be reduced based on what is left to wait for.

Reply:
That seems an issue with the methods preexisting logic. Should we look into this on a different ticket, or should this be fixed here?